aws/rust-runtime/aws-config/src/default_provider.rs

```diff
@@ -347,5 +444,129 @@ pub mod credentials {
                 .expect("creds should load");
             assert_eq!(creds.access_key_id(), "correct_key_secondary");
         }
+
+        fn fs_from_tuples(tuples: &[(&str, &str)]) -> Fs {
+            let mut fs = HashMap::new();
+            tuples.iter().for_each(|(k, v)| {
+                let k = (*k).to_owned();
+                let v = v.as_bytes().to_vec();
+                fs.insert(k, v);
+            });
+            Fs::from_map(fs)
+        }
+
+        #[tokio::test]
+        async fn test_returns_default_retry_config_from_empty_profile() {
+            let env = Env::from_slice(&[("AWS_CONFIG_FILE", "config")]);
+            let fs = fs_from_tuples(&[("config", "[default]\n")]);
+
+            let provider_config = ProviderConfig::no_configuration().with_env(env).with_fs(fs);
+
+            let actual_retry_config = retry_config::default_provider()
+                .configure(&provider_config)
+                .retry_config()
+                .await;
+
+            let expected_retry_config = RetryConfig::new();
+
+            assert_eq!(actual_retry_config, expected_retry_config)
+        }
```

Comment:
can you have one of these test assert that we're actually getting the default values we're expecting?